general/session.md

```diff
@@ -0,0 +1,26 @@
+# Session API
+
+## Workflow
+
+### Getting
+
+When getting a value stored in Session, we are deserializing it by removing the `serialize://` string from the data we retrieve. After the data is deserialized we get access to the data requested, stripped of any serialization.
+
+#### Needs clarifying
+
+-   Even though we are using POST we are sending a `'cmd': 'get'`, which might be revealing the underlying architecture of how Liferay handles fetch requests.
+-   Why are we using a POST method in a getter?
+
+### Setting
+
+When setting values to Session, we have to serialize them first so that it is served in a proper way that the endpoint expects. After the value has been serialized, we append it to FormData which then get's sent to the `session_click` endpoint to be consumed at a later time.
+
+## Components used by the Session API
+
+### URL
+
+The base URL used by the Session API is `http://localhost:8080/c/portal/session_click` where `http://localhost:8080/` and `c/` are different depending on the server. This URL is appended by FormData, which consists of `'cmd': 'get'` (when using `getSessionValue`) or `'cmd': 'set'` (when using `setSessionValue`), and `'p_auth: Liferay.authToken`
```

Comment:
I don't think any of this matters. If somebody is interested about it, they can look at the source. The purpose of these docs is above all else to:

1. Provide a guide for usage (probably pretty trivial if we're talking about methods which take params `key` and `key, data`, respectively).
2. Give a high-level sense of what this API is for, conceptually.

I don't think translating the implementation from JS to English is really worth the time and effort though (at best, it's work you didn't really need to do because you could have linked to the functions which are both very short, and at worst, it's inaccurate).